addons/web_editor/static/src/js/wysiwyg/widgets/color_palette.js

```diff
@@ -278,18 +318,21 @@ const ColorPaletteWidget = Widget.extend({
      *
      * @param {string|number} ccValue
      * @param {string} color rgb[a]
+     * @param {boolean} [selectTab=true]
      */
-    setSelectedColor: function (ccValue, color) {
+    setSelectedColor: function (ccValue, color, selectTab = true) {
```

Comment:
I don't understand why this is needed

Reply:
This is to avoid that when `setSelectedColor` is called from the previous comment's location (`ColorpickerUserValueWidget.setValue()`) the tab gets reset to the initially defined one. (Because when CCs are available, fp requested that it re-opens on the CC tab, and because of the optimization you mentioned which avoids the 100s of color palettes from being created before accessing them when clicking on the THEME tab)

Reply:
Explain that near the callers ;)